### PR TITLE
feat: implement ICP verfiable Credential with Dacade as relying party

### PR DESCRIPTION
**Pull request by Jonath-z**
# Submit a pull request

- [x] This is not a duplicate of an existing pull request.
- [x] No existing features have been broken without good reason.
- [x] Your commit messages are detailed
- [x] The code style guideline have been followed.
- [x] Documentation has been updated to reflect your changes.
- [ ] Tests have been added or updated to reflect your changes.
- [ ] All tests pass.

---

Replace any ":question:" below with information about your pull request.

## Pull Request Details
This PR implements ICP verifiable Credential for ICP courses completion with Dacade as relying party.

## How to test
For now the test need to happen locally due module resolution incompatibity of the Verifiable Credential SDK and Next.js mendatory module resolution.

These are the step to follow locally:
1. Make sure to have node 22 and above
2. Run the app locally
3. Navigate to an ICP achievement
4. Click Request Credential button 

> Note: This demo runs on ICP non official identity provider and uses a testing credential issuer

More info:
- https://internetcomputer.org/docs/current/developer-docs/identity/verifiable-credentials/overview
- https://internetcomputer.org/docs/current/developer-docs/identity/verifiable-credentials/how-it-works#test-verifiable-credentials

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -9,7 +9,7 @@ jobs:
       - uses: actions/checkout@v4
       - uses: actions/setup-node@v4
         with:
-          node-version: "20.0.0"
+          node-version: "22.6.0"
```
Comment (musayann) [inline .github/workflows/ci.yml:12]: node `22` is not fully supported by many node modules yet. Especially next js

File: src/pages/achievements/[id].tsx [hunk 2]
```diff
@@ -96,14 +99,38 @@ const Achievement = () => {
...
   }, [achievement]);
 
+  const addCourseCompletionToTheIssuerCanister = useCallback(async () => {
+    if (!achievement?.metadata) return;
+    const { name } = achievement?.metadata;
+    if (!window.issuerCanister) return;
+    try {
+      setAddCourseToCompletionPending(true);
+      await window.issuerCanister.add_course_completion(name.toLowerCase().replaceAll(" ", "-"));
+      await dispatch(completeIcpCertificate({ id: achievement.id }));
```
Comment (musayann) [inline src/pages/achievements/[id].tsx:121]: so if this api endpoint gets triggered certificate be marked as completed even if icp hasn't been triggered

**Review by musayann — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml (hunk 1 above)
Comment (Jonath-z) [reply .github/workflows/ci.yml:12]: Unfortunately, it's needed for some ICPs peer dependencies to work

File: src/pages/achievements/[id].tsx (hunk 2 above)
Comment (Jonath-z) [reply src/pages/achievements/[id].tsx:121]: Yes, but the badge can never be verified with badge gate or any other, as badge gate does not communicate with dacade backend.

**Review by musayann — APPROVED** [review]

**Pull request merged**